packages/java-edition/src/json/checker/index.ts

```diff
@@ -1,66 +1,25 @@
-import { localize } from '@spyglassmc/locales'
 import { dissectUri } from '../../binder/index.js'
 
-const Checkers = new Map<core.FileCategory, `::${string}::${string}`>([
-	['advancement', '::java::data::advancement::Advancement'],
-	['dimension', '::java::data::worldgen::dimension::Dimension'],
-	['dimension_type', '::java::data::worldgen::dimension::DimensionType'],
-	['item_modifier', '::java::data::item_modifier::ItemModifier'],
-	['loot_table', '::java::data::loot::LootTable'],
-	['predicate', '::java::data::predicate::Predicate'],
-	['recipe', '::java::data::recipe::Recipe'],
-	['worldgen/biome', '::java::data::worldgen::biome::Biome'],
-	[
-		'worldgen/configured_carver',
-		'::java::data::worldgen::carver::ConfiguredCarver',
-	],
-	[
-		'worldgen/configured_surface_builder',
-		'::java::data::worldgen::surface_builder::ConfiguredSurfaceBuilder',
-	],
-	['worldgen/configured_feature', '::java::data::feature::ConfiguredFeature'],
-	[
-		'worldgen/configured_structure_feature',
-		'::java::data::worldgen::structure::Structure',
-	],
-	[
-		'worldgen/density_function',
-		'::java::data::worldgen::density_function::DensityFunction',
-	],
-	[
-		'worldgen/noise',
-		'::java::data::worldgen::dimension::biome_source::NoiseParameters',
-	],
-	[
-		'worldgen/noise_settings',
-		'::java::data::worldgen::noise_settings::NoiseGeneratorSettings',
-	],
-	[
-		'worldgen/processor_list',
-		'::java::data::worldgen::processor_list::ProcessorList',
-	],
-	[
-		'worldgen/template_pool',
-		'::java::data::worldgen::template_pool::TemplatePool',
-	],
-])
-
-export const file: core.Checker<json.JsonFileNode> = (node, ctx) => {
-	const child = node.children[0]
+export const entry: core.Checker<json.JsonNode> = (node, ctx) => {
+	if (ctx.doc.uri.endsWith('/pack.mcmeta')) {
+		return json.checker.definition('::java::pack::Pack')(node, ctx)
+	}
 	const parts = dissectUri(ctx.doc.uri, ctx)
-	if (parts && Checkers.has(parts.category)) {
-		const identifier = Checkers.get(parts.category)!
-		return json.checker.definition(identifier)(child, ctx)
-	} else if (parts?.category.startsWith('tag/')) {
-		// TODO
-		return json.checker.definition('::java::data::tag::Tag')(child, ctx)
-	} else if (ctx.doc.uri.endsWith('/pack.mcmeta')) {
-		return json.checker.definition('::java::pack::Pack')(child, ctx)
+	if (parts) {
+		if (parts?.category.startsWith('tag/')) {
+			return json.checker.resource('tag')(node, ctx)
+		}
+		return json.checker.resource(parts.category)(node, ctx)
 	}
 }
 
 export function register(meta: core.MetaRegistry) {
-	meta.registerChecker<json.JsonFileNode>('json:file', file)
+	meta.registerChecker<json.JsonNode>('json:array', entry)
+	meta.registerChecker<json.JsonNode>('json:boolean', entry)
+	meta.registerChecker<json.JsonNode>('json:null', entry)
+	meta.registerChecker<json.JsonNode>('json:number', entry)
+	meta.registerChecker<json.JsonNode>('json:object', entry)
+	meta.registerChecker<json.JsonNode>('json:string', entry)
```

Comment:
Looks like a merge conflict resolved the wrong way. It should only handle `json:file` nodes now.

Reply:
I couldn't resolve this without errors, I'm not sure how to handle this properly, I'd appreciate a fix, can't hop in VC this week.